pressbooks/pressbooks-build-tools
# feat: upgrade to Laravel Mix 6.x

This PR updates Laravel Mix to 6.0 and brings a number of other dependencies up to date as well.

For testing purposes, I've published this PR to NPM as [3.0.0-canary.1](https://www.npmjs.com/package/pressbooks-build-tools/v/3.0.0-canary.1). It can be installed using the `next` tag, e.g.:

```bash
npm install --no-save pressbooks-build-tools@next
```

### Testing

To test in [pressbooks/pressbooks](https://github.com/pressbooks/pressbooks/):

1. Install as directed above.
2. Modify NPM build scripts [as directed in the README](https://github.com/greatislander/pressbooks-build-tools/blob/5b74ac33338bbe7d75b3b8e139d78ea29a848bb5/README.md#upgrading-to-30).
3. Test build and lint scripts to make sure that they work as expected.

To test in CI:

1. Create a branch of [pressbooks/pressbooks](https://github.com/pressbooks/pressbooks/).
2. Modify [this line](https://github.com/pressbooks/pressbooks/blob/1457ab067706bf0ccb92fa3560a65ecac0cb010b/.github/workflows/phpunit.yml#L59) to:
   ```
   npm install --no-save pressbooks-build-tools@next
   ```
3. Open a pull request to dev to verify that tests pass.

### After Testing

If testing produces the expected results, this PR can be merged.

Then:

1. Update the version in package.json to `3.0.0`.
2. Run `npm install`.
3. Run `npm publish` and supply the one-time code for NPM (I think @ho-man-chan is set up to do this).

Commit the changes to `package.json` and tag 3.0.0.

This will allow the existing `install-build-tools` to be used to install the updated version.
